File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-10.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l = 0;
        int res = 0;
        set<int> seen;

        for(int r = 0; r < s.size(); r++){
            while(seen.find(s[r]) != seen.end()){
                seen.erase(s[l]);
                l++;         
            }
            res = max(res, r-l+1);
            seen.insert(s[r]);
        }
        return res;
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-10.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        res = nums[0]
        l, r = 0, len(nums)-1


        while l <= r:
            m = (l + r) // 2
            print(m)

            if nums[l] <= nums[r]:
                res = min(res, nums[l])
                break

            if nums[m] >= nums[l]:
                l = m+1
            else:
                res = nums[m]
                r = m-1

        return res